helm/cluster-vsphere/templates/machinedeployment.yaml

```diff
@@ -1,4 +1,9 @@
-{{- range $name, $value := .Values.global.nodePools }}
+{{- range $name, $value := include "createMapOfWorkerPoolSpecs" . | fromYaml }}
+{{- $c := (merge (dict "currentPool"  $value) $.Values) }}
+{{/* we deepcopy the $value variable inside the loop because it is mutated when it is passed to
+kubeadmConfigTemplateRevision which then affects later functions which also rely on $values.
+$valuesDeepCopied should only be used with the call to kubeadmConfigTemplateRevision */}}
+{{- $valueDeepCopied := (deepCopy $value) -}}
```

Comment:
deepCopying the values is necessary because they are mutated on line 31 which breaks the templating on line 37